test/integration_test.go

```diff
@@ -1652,7 +1652,29 @@ func TestDependencyOutputTypeConversion(t *testing.T) {
 	assert.Equal(t, outputs["object"].Value, map[string]interface{}{"list": []interface{}{1.0, 2.0, 3.0}, "map": map[string]interface{}{"foo": "bar"}, "num": 42.0, "str": "string"})
 	assert.Equal(t, outputs["string"].Value, "string")
 	assert.Equal(t, outputs["from_env"].Value, "default")
+}
+
+// Test that we get the expected error message about dependency cycles when there is a cycle in the dependency chain
+func TestDependencyOutputCycleHandling(t *testing.T) {
```

Comment:
What type of cycle does this test? I'd want to check:

A -> B -> A

A -> B -> C -> D -> A

A -> A

Reply:
Good point. Added each of these cases and another one:

A -> B -> A
 |
 -> C